### PR TITLE
feat(user registration) : create event rule and attach it to SNS target + eventSchema

### PR DESCRIPTION
**Pull request by sri-kirsh**
## Goal
create event rule and attach it to SNS target

todo: not creating DLQ alert yet coz we have a [chores ticket to fix the general alert pattern](https://getpocket.atlassian.net/browse/INFRA-1048)

### dev deployment:
- [event rule and sns target](https://us-east-1.console.aws.amazon.com/events/home?region=us-east-1#/eventbus/PocketEventBridge-Dev-Shared-Event-Bus/rules/PocketEventBridge-Dev-UserRegistration-Rule-Rule)
- [event schema ](https://us-east-1.console.aws.amazon.com/events/home?region=us-east-1#/registries/PocketEventBus/schemas/User-Registration-Event)

Ticket: https://getpocket.atlassian.net/browse/UJM-32

### CODE REVIEW

File: .aws/src/event-rules/user-registration/userRegistrationEventRule.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+    this.snsTopic = new sns.SnsTopic(this, 'user-registration-topic', {
+      name: `${config.prefix}-UserRegistrationTopic`,
+      lifecycle: {
+        preventDestroy: true,
+      },
+    });
+
+    this.snsTopicDlq = new sqs.SqsQueue(this, 'sns-topic-dql', {
+      name: `${config.prefix}-SNS-Topic-Event-Rule-DLQ`,
```
Comment (jpetto) [inline .aws/src/event-rules/user-registration/userRegistrationEventRule.ts:34]: should this reference user registration? like `${config.prefix}-UserRegistrationTopic-DLQ`?

**Review by jpetto — APPROVED** [review]
one question about naming conventions, but doesn't need to block if it's not pertinent.

File: .aws/src/event-rules/user-registration/userRegistrationEventRule.ts (hunk 1 above)
Comment (sri-kirsh) [reply .aws/src/event-rules/user-registration/userRegistrationEventRule.ts]: thanks for catching that @jpetto  :) 
looks like c-p error, we have similar names in other events too - updated them all

File: .aws/src/event-rules/user-registration/eventConfig.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { config as globalConfig } from '../../config';
+
+export const eventConfig = {
+  name: 'UserRegistration',
+  source: 'web-repo',
+  detailType: ['User Registration'],
```
Comment (sri-kirsh) [inline .aws/src/event-rules/user-registration/eventConfig.ts:6]: should match to: https://github.com/Pocket/Web/blob/179947cbae91f4451094906a34b589efcdcc09dd/tests/functional/AWS/EventBridge/SubscriberTest.php#L98

File: .aws/src/events-schema/userRegistrationEventSchema.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  private getUserRegistrationPayload() {
+    return {
+      openapi: '3.0.0',
+      info: {
+        version: '1.0.0',
+        title: 'Event',
+      },
+      paths: {},
+      components: {
+        schemas: {
```
Comment (sri-kirsh) [inline .aws/src/events-schema/userRegistrationEventSchema.ts:48]: e.g event emitter payload here: https://github.com/Pocket/Web/pull/4529/files#diff-cadfbdab536c5132941c11747f66f205b9f3b7793128c3f3e69375e92bd9fbbc

**Review by jpetto — APPROVED** [review]

**Pull request merged**